File: src/test/java/org/projectodd/nodej/AssertTest.java
package org.projectodd.nodej;

import static org.fest.assertions.Assertions.assertThat;

import org.dynjs.exception.ThrowException;
import org.dynjs.runtime.JSObject;
import org.dynjs.runtime.Types;
import org.junit.Test;

public class AssertTest extends NodejTestSupport {

    @Test
    public void testAssertOk() {
        // if != we get an error
        assertThat(eval("require('assert').ok(true)")).isEqualTo(Types.UNDEFINED);
    }
    
    @Test
    public void testAssertFail() {
        try {
            eval("require('assert').fail('actual', 'expected', 'message', null)");
            throw new AssertionError("assert.fail() should have thrown an AssertionError");
        } catch (ThrowException e) {
            JSObject err = (JSObject) e.getValue();
            assertThat(err.get(null, "name")).isEqualTo("AssertionError");
        }
    }
}

File: src/test/resources/assert/assert_test.js
var vertxTest = require("vertx_tests");
var vassert   = vertxTest.vassert;

assert = require('assert');

var AssertTest = {
  testAssertFail: function() {
    try {
      assert.fail('actual', 'expected', 'message', '=');
      vassert.fail("Node assert fail should throw.");
    } catch(e) {
      vassert.assertTrue(e instanceof assert.AssertionError);
      vassert.assertEquals('message', e.message);
      vassert.assertEquals('actual', e.actual);
      vassert.assertEquals('expected', e.expected);
      vassert.assertEquals('=', e.operator);
    }
    vassert.testComplete();
  },

  testAssertOK: function() {
    assert.ok(true);
    try {
      assert.ok(false, 'a message');
    } catch(err) {
      vassert.assertEquals('a message', err.message);
    }
    vassert.testComplete();
  },

  testAssert: function() {
    assert(true);
    try {
      assert(false, 'a message');
    } catch(err) {
      vassert.assertEquals('a message', err.message);
    }
    vassert.testComplete();
  },

  testAssertEqual: function() {
    assert.equal('1', '1');
    try {
      assert.equal('1', '2');
    } catch(err) {
      vassert.assertEquals('"1" == "2"', err.message);
    }
    vassert.testComplete();
  },

  testAssertNotEqual: function() {
    assert.notEqual('1', '2');
    try {
      assert.notEqual('1', '1');
    } catch(err) {
      vassert.assertEquals('"1" != "1"', err.message);
    }
    vassert.testComplete();
  }
}


vertxTest.startTests(AssertTest);
